app/main.py

```diff
@@ -1,3 +1,49 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, number: any) -> float:
+        if isinstance(number, Distance):
+            return Distance(self.km + number.km)
+        else:
```

Comment:
you don`t have to write else after return. this block is not reachable if isinstance(number, Distance) return True

Reply:
the same applies to each else block below

Reply:
I would also recommend checking for type errors like:
self.km + "a string or another unsupportable type"

Reply:
Yep, that is correct. Fix everywhere